packages/react-components/react-menu/library/src/components/MenuSplitGroup/useMenuSplitGroup.ts

```diff
@@ -54,17 +56,50 @@ export const useMenuSplitGroup_unstable = (
     components: {
       root: 'div',
     },
+    setMultiline,
     root: slot.always(
       getIntrinsicElementProps('div', {
         role: 'group',
         // FIXME:
         // `ref` is wrongly assigned to be `HTMLElement` instead of `HTMLDivElement`
         // but since it would be a breaking change to fix it, we are casting ref to it's proper type
-        ref: useMergedRefs(ref, innerRef) as React.Ref<HTMLDivElement>,
+        ref: useMergedRefs(ref, innerRef, multilineRef) as React.Ref<HTMLDivElement>,
         onKeyDown,
         ...props,
       }),
       { elementType: 'div' },
     ),
   };
 };
+
+/**
+ * Creates a callback that lets a multiline menu item child set an attribute on this component
+ * Children can mount before parents so we need to store the value and apply it when the parent is mounted
+ */
+const useHandleMultilineMenuItem = () => {
+  const [[multilineRef, setMultiline]] = React.useState(() => {
+    let isMultiline = false;
+    let multilineNode: HTMLElement | null = null;
+
+    function applyAttr() {
+      multilineNode?.toggleAttribute(menuSplitGroupMultilineAttr, isMultiline);
+    }
+
+    return [
+      (node: HTMLDivElement | null) => {
+        if (node) {
+          multilineNode = node;
+          applyAttr();
+        } else {
+          multilineNode = null;
+        }
+      },
+      (value: boolean) => {
+        isMultiline = value;
+        applyAttr();
+      },
+    ] as const;
+  });
+
+  return { multilineRef, setMultiline };
```

Comment:
```suggestion
  const [handle] = React.useState(() => {
    let isMultiline = false;
    let multilineNode: HTMLElement | null = null;

    function applyAttr() {
      multilineNode?.toggleAttribute(menuSplitGroupMultilineAttr, isMultiline);
    }

    return {
      multilineRef: (node: HTMLDivElement | null) => {
        if (node) {
          multilineNode = node;
          applyAttr();
        } else {
          multilineNode = null;
        }
      },
      setMultiline: (value: boolean) => {
        isMultiline = value;
        applyAttr();
      },
    };
  });

  return handle;
```

Oops, overlooked that this will work better in this case